sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraServiceImpl.java

```diff
@@ -284,12 +287,16 @@ private Cluster getCluster(
       int port,
       String username,
       String password,
+      PasswordDecrypter passwordDecrypter,
       String localDc,
       String consistencyLevel) {
     Cluster.Builder builder =
         Cluster.builder().addContactPoints(hosts.toArray(new String[0])).withPort(port);
 
     if (username != null) {
+      if (passwordDecrypter != null) {
+        password = passwordDecrypter.decrypt(password);
```

Comment:
@echauchot You're right, currently the password neither encrypted nor decrypted.
I understand your determination to not expose additional parameters. I can provide a default password decrypter, such an encryption system based on a key (AES, RSA, ...) but in this case and without configuration, the (private) key will also be serialized in the pipeline.
Therefore, with any encryption system, we must configure an external parameter, otherwise the encryption information will be serialized in the pipeline.

Reply:
mmm, good point ! In that case, let's go for providing the password decrypter; it is better to add a configuration knob than serializing a confidential information (password or private key). 
Owing to your name I suppose you are french so I'd say: "entre deux maux choisissons le moindre" :)

Reply:
Also I'm waiting for one of my other PRs to be merged, it is a complete refactoring that also adds an embedded real Cassandra to UTests. The better would be to add your test as a UTest rather than an ITest. In beam we do as much as possible in Utest, ITests are more for cross-runners tests and load testing

Reply:
Okay, understood. I will wait your [PR](https://github.com/apache/beam/pull/7751) will be merged to rewrite the related UTest

Reply:
perfect ! Trying to get an available reviewer for my PR right now

Reply:
@mblmat I just merged my [PR](https://github.com/apache/beam/pull/7751), please rebase, refactor (there is no more CassandraService and CassandraServiceImpl) and adapt your UTests to the new embedded Cassandra instance